tests/test_model_analyzer.py

```diff
@@ -0,0 +1,253 @@
+"""
+Test cases to verify that ModelAnalyzer works correctly
+on actual scikit-agent models.
+"""
+
+from importlib.util import find_spec
+import pytest
+import sys
+
+sys.path.append("../src")
+
+SKAGENT_AVAILABLE = find_spec("skagent") is not None
+HAS_DEPENDENCIES = SKAGENT_AVAILABLE
```

Comment:
skagent isn't an optional dependency. It is the package that is being tested here.